src/language/from_relay/mod.rs

```diff
@@ -1808,4 +1813,306 @@ def @main(%x: Tensor[(3), float32]) -> Tensor[(3), float32] {
+                        // generate somewhat realistic image data
+                        index += 1;
+                        ndarray::ArrayD::<f32>::random_using(
+                            shape.clone(),
+                            Uniform::new(0f32, 255f32),
+                            &mut tensor_rng,
+                        )
+                    } else {
+                        // generate parameters
+                        ndarray::ArrayD::<f32>::random_using(
+                            shape.clone(),
+                            Uniform::new(-1f32, 1f32),
+                            &mut tensor_rng,
+                        )
+                    };
+                    env.insert(name.as_str(), value.clone());
+                    let filepath = std::env::temp_dir().with_file_name(format!(
+                        "arg-{}.npy",
+                        rand::thread_rng()
+                            .sample_iter(&rand::distributions::Alphanumeric)
+                            .take(30)
+                            .collect::<String>()
+                    ));
+                    write_npy(&filepath, &value).unwrap();
+                    cmd.arg(filepath);
+                }
+
+                let mut proc = cmd.spawn().ok().expect("Failed to spawn process");
+                proc.stdin
+                    .as_mut()
+                    .unwrap()
+                    .write_all(relay_str.as_bytes())
+                    .unwrap();
+                let output = proc.wait_with_output().unwrap();
+                // Check that it ran.
+                assert!(
+                    output.status.success(),
+                    "Running Relay code failed with code {:?}.\nstdout:\n{}\nstderr:\n{}",
+                    output.status.code(),
+                    std::str::from_utf8(output.stdout.as_slice())
+                        .expect("Could not convert stderr to UTF8"),
+                    std::str::from_utf8(output.stderr.as_slice())
+                        .expect("Could not convert stderr to UTF8")
+                );
+
+                // TODO(@gussmith23) output type assumption
+                let relay_output: ndarray::ArrayD<f32> = read_npy(output_filepath).unwrap();
+
+                b.iter(|| {
+                    // use black box to prevent compiler optimizations
+                    let expr = test::black_box(&expr);
+                    let env = test::black_box(&env);
+
+                    let interpreter_output = match interpret(&expr, expr.as_ref().len() - 1, &env) {
+                        crate::language::interpreter::Value::Access(a) => a.tensor,
+                        _ => panic!(),
+                    };
+
+                    // TODO: this was a hack because NAN != NAN
+                    // relay_output.mapv_inplace(|x| if x.is_nan() { 0.0 } else { x });
+                    // interpreter_output.mapv_inplace(|x| if x.is_nan() { 0.0 } else { x });
```

Comment:
Should we consider adding a check to assert that there are no NaN values? It seems like there shouldn't be any.

Reply:
NaN values don't appear in the shallow model testing, but could in the full model (i will test this again...)

Reply:
i think we can come back to this when I do memo + ndarray since its hard to tell currently whether there are Nan values or not :\

Reply:
Would something like
```rust
interpreter_output.iter().all(|v| !v.is_nan())
```
work?